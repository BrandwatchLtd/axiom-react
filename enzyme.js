// Prevent React 16 warnings on the console during tests using requestAnimationFrame
global.requestAnimationFrame = setTimeout;

import Enzyme from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";

Enzyme.configure({ adapter: new Adapter() });
