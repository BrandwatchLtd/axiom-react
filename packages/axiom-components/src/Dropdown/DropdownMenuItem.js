import PropTypes from 'prop-types';
import React, { Component } from 'react';
import omit from 'lodash.omit';
import ContextMenuItem from '../Context/ContextMenuItem';

export default class DropdownMenuItem extends Component {
  static propTypes = {
    /** SKIP */
    children: PropTypes.node,
    closeDropdown: PropTypes.func,
    /** Disabled interactions and applies styling */
    disabled: PropTypes.bool,
   /**
    * Whether the menu should stay open after being clicked.
    */
    keepOpen: PropTypes.bool,
     /**
     * Whether the menu is part of a multi-selection menu. When true, this
     * will disable automatically closing the DropdownMenu when clicked.
     */
    multiSelect: PropTypes.bool,
    /** Click handler  */
    onClick: PropTypes.func,
    /** Provides indication that the menu item is selected */
    selected: PropTypes.bool,
  };

  constructor(props) {
    super(props);
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick(event) {
    const { closeDropdown, multiSelect, keepOpen, onClick } = this.props;

    if (closeDropdown && !multiSelect && !keepOpen) closeDropdown();

    if (onClick) onClick(event);
  }

  render() {
    const { children, closeDropdown, ...rest } = this.props;

    return (
      <ContextMenuItem
          { ...omit(rest, ['keepOpen']) }
          onClick={ event => this.handleClick(event) }
          tabIndex="0">
        { children }
      </ContextMenuItem>
    );
  }
}
