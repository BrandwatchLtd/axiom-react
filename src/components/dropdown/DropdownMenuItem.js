import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { ContextMenuItem } from 'bw-axiom';

export default class DropdownMenuItem extends Component {
  static propTypes = {
    children: PropTypes.node,
    onClick: PropTypes.func,
  };

  static contextTypes = {
    closeDropdown: PropTypes.func,
  };

  handleClick(...args) {
    const { closeDropdown } = this.context;
    const { onClick } = this.props;

    if (closeDropdown) {
      closeDropdown();
    }

    if (onClick) {
      onClick(...args);
    }
  }

  render() {
    const { children, ...rest } = this.props;

    return (
      <ContextMenuItem { ...rest } onClick={ this.handleClick.bind(this) }>
        { children }
      </ContextMenuItem>
    );
  }
}
