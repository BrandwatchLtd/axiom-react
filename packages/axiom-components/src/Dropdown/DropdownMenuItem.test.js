import React from 'react';
import { shallow } from 'enzyme';
import DropdownMenuItem from './DropdownMenuItem';
import toJson from 'enzyme-to-json';

const render = (props = {}, opts = {}) => {
  return shallow(
    <DropdownMenuItem { ...props }>
      Test
    </DropdownMenuItem>, opts);
} ;

const wrapElementWithContext = (element, context) => {
  const Children = element.props().children(context);
  return shallow(Children);
};

describe('DropdownMenuItem', () => {
  let props;
  let opts;
  let mockEvent;

  beforeEach(() => {
    props = {
      disabled: false,
      keepOpen: false,
      multiSelect: false,
    };
    mockEvent = {};
  });

  describe('onClick', () => {
    let mockCloseDropdown;
    let context;

    beforeEach(() => {
      mockCloseDropdown = jest.fn();
      context = { closeDropdown: mockCloseDropdown };
    });

    it('closes the dropdown menu when clicked', () => {
      const element = render(props, opts);
      const elementWithContext = wrapElementWithContext(element, context);
      elementWithContext.simulate('click', mockEvent);
      expect(mockCloseDropdown).toHaveBeenCalledTimes(1);
    });

    it('does not close the dropdown when multiSelect is set', () => {
      props.multiSelect = true;
      const element = render(props, opts);
      const elementWithContext = wrapElementWithContext(element, context);
      elementWithContext.simulate('click', mockEvent);
      expect(mockCloseDropdown).not.toHaveBeenCalled();
    });

    it('does not close the dropdown when keepOpen is set', () => {
      props.keepOpen = true;
      const element = render(props, opts);
      const elementWithContext = wrapElementWithContext(element, context);
      elementWithContext.simulate('click', mockEvent);
      expect(mockCloseDropdown).not.toHaveBeenCalled();
    });
  });

  describe('Snapshots', () => {
    it('renders with defaultProps', () => {
      expect(toJson(wrapElementWithContext(render()))).toMatchSnapshot();
    });

    it('renders with disabled', () => {
      expect(toJson(wrapElementWithContext(render({
        disabled: true,
      }, opts)))).toMatchSnapshot();
    });

    it('renders with index', () => {
      expect(toJson(wrapElementWithContext(render({
        index: '1',
      }, opts)))).toMatchSnapshot();
    });

    it('renders with keepOpen', () => {
      expect(toJson(wrapElementWithContext(render({
        keepOpen: true,
      }, opts)))).toMatchSnapshot();
    });

    it('renders with multiSelect', () => {
      expect(toJson(wrapElementWithContext(render({
        multiSelect: true,
      }, opts)))).toMatchSnapshot();
    });

    it('renders with onClick', () => {
      expect(toJson(wrapElementWithContext(render({
        onClick: () => {},
      }, opts)))).toMatchSnapshot();
    });

    it('renders with onMouseOver', () => {
      expect(toJson(wrapElementWithContext(render({
        onMouseOver: () => { },
      }, opts)))).toMatchSnapshot();
    });
  });
});
