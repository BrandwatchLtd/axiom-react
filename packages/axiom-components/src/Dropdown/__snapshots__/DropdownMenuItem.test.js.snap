// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownMenuItem Snapshots renders with defaultProps 1`] = `
<ContextMenuItem
  onClick={[Function]}
  tabIndex="0"
>
  Test
</ContextMenuItem>
`;

exports[`DropdownMenuItem Snapshots renders with disabled 1`] = `
<ContextMenuItem
  disabled={true}
  onClick={[Function]}
  tabIndex="0"
>
  Test
</ContextMenuItem>
`;

exports[`DropdownMenuItem Snapshots renders with index 1`] = `
<ContextMenuItem
  index="1"
  onClick={[Function]}
  tabIndex="0"
>
  Test
</ContextMenuItem>
`;

exports[`DropdownMenuItem Snapshots renders with keepOpen 1`] = `
<ContextMenuItem
  onClick={[Function]}
  tabIndex="0"
>
  Test
</ContextMenuItem>
`;

exports[`DropdownMenuItem Snapshots renders with multiSelect 1`] = `
<ContextMenuItem
  multiSelect={true}
  onClick={[Function]}
  tabIndex="0"
>
  Test
</ContextMenuItem>
`;

exports[`DropdownMenuItem Snapshots renders with onClick 1`] = `
<ContextMenuItem
  onClick={[Function]}
  tabIndex="0"
>
  Test
</ContextMenuItem>
`;

exports[`DropdownMenuItem Snapshots renders with onMouseOver 1`] = `
<ContextMenuItem
  onClick={[Function]}
  onMouseOver={[Function]}
  tabIndex="0"
>
  Test
</ContextMenuItem>
`;
